adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -873,10 +865,11 @@
         function checkLoginIframe() {
             var promise = createPromise();
 
-            if (loginIframe.iframe && loginIframe.iframeOrigin) {
+            if (loginIframe.iframe && loginIframe.iframeOrigin && loginIframe.callbackList.length !== 0) {
                 var msg = {};
                 msg.callbackId = createCallbackId();
-                loginIframe.callbackMap[msg.callbackId] = promise;
+                msg.sessionId = kc.sessionId;
+                loginIframe.callbackList.push(promise);
```

Comment:
But isn't it the same with the current check I added in the `if` ? And in the `else` the promise is resolved.